app/tests/routers/test_data_decoder.py

```diff
@@ -1,33 +1,46 @@
-import unittest
+from typing import cast
 
 from fastapi.testclient import TestClient
 
 from hexbytes import HexBytes
+from safe_eth.eth.constants import NULL_ADDRESS
+from safe_eth.eth.utils import get_empty_tx_params
 from sqlmodel.ext.asyncio.session import AsyncSession
+from web3 import Web3
+from web3.types import ABI
 
 from ...datasources.db.database import database_session
-from ...datasources.db.models import AbiSource
+from ...datasources.db.models import Abi, AbiSource, Contract
 from ...main import app
 from ...services.abis import AbiService
+from ...services.data_decoder import get_data_decoder_service
+from ..datasources.db.db_async_conn import DbAsyncConn
 
 
-class TestRouterAbout(unittest.TestCase):
+class TestRouterAbout(DbAsyncConn):
     client: TestClient
 
     @classmethod
     def setUpClass(cls):
         cls.client = TestClient(app)
 
+    def setUp(self):
```

Comment:
Should uses async setup and async teardown?

Reply:
If the functions inside are not async, I wouldn't